mslib/utils/config.py

```diff
@@ -641,6 +663,11 @@ def merge_dict(existing_dict, new_dict):
             data, match = compare_data(existing_dict[key], new_dict[key])
             if match:
                 existing_dict[key] = data
+                
+                # Add special validation for data_dir
+                if key == "data_dir" and not validate_data_dir(data):
+                    logging.warning(f"Invalid data_dir format: {data}. Must start with '~' or a filesystem URL.")
+                    existing_dict[key] = existing_dict.get("data_dir", "~/mssdata")
```

Comment:
this will overwrite a user definition. I think we should not do that, but give the hint what is wrong is good.

Reply:
The problem with this approach is, someone could have already a different fsurl as default and then did a change which will remove his dir and reset to our default. 

`existing_dict[key] = existing_dict.get("data_dir", "~/mssdata")`